packages/router/src/grpc/view-protocol-server/nullifier-status.ts

```diff
@@ -33,26 +29,28 @@ export const nullifierStatus: Impl['nullifierStatus'] = async (req, ctx) => {
   // zero-height spend should never appear. So if one of these is truthy, the
   // nullifier is spent.
   const [swap, note] = await Promise.all([
-    indexedDb.getSwapByNullifier(nullifier),
-    indexedDb.getSpendableNoteByNullifier(nullifier),
+    indexedDb.getSwapByNullifier(findNullifier),
+    indexedDb.getSpendableNoteByNullifier(findNullifier),
   ]);
-  const spent = Boolean(swap?.heightClaimed) || Boolean(note?.heightSpent);
-
-  if (!spent && req.awaitDetection) {
-    // use of the nullifier was not present in db, so watch that subscription.
-    // we might double-check very recent items, but we won't miss any.
-    const eventuallySpent = Promise.race([
-      watchStream(swapStream, ({ value: swapJson }) => {
-        const swap = SwapRecord.fromJson(swapJson);
-        return Boolean(swap.heightClaimed) && nullifier.equals(swap.nullifier);
-      }),
-      watchStream(noteStream, ({ value: noteJson }) => {
-        const note = SpendableNoteRecord.fromJson(noteJson);
-        return Boolean(note.heightSpent) && nullifier.equals(note.nullifier);
-      }),
-    ]);
-    return eventuallySpent.then(() => ({ spent: true }));
-  }
+
+  const spent =
+    Boolean(swap?.heightClaimed) ||
+    Boolean(note?.heightSpent) ||
+    Boolean(
+      awaitDetection &&
+        // use of the nullifier was not present in db, so watch that subscription.
+        // we might double-check very recent items, but we won't miss any.
+        (await Promise.race([
+          watchSubscription(swapStream, ({ value: swapJson }) => {
+            const swap = SwapRecord.fromJson(swapJson);
+            return Boolean(swap.heightClaimed) && findNullifier.equals(swap.nullifier);
+          }),
+          watchSubscription(noteStream, ({ value: noteJson }) => {
+            const note = SpendableNoteRecord.fromJson(noteJson);
+            return Boolean(note.heightSpent) && findNullifier.equals(note.nullifier);
+          }),
+        ])),
+    );
```

Comment:
A bit hard to follow. I'd pull out this bit into its own variable above `const spent`, and then refer to that variable here.